components/director/internal/domain/formation/notifications.go

```diff
@@ -148,19 +154,52 @@ func (ns *notificationsService) GenerateNotifications(ctx context.Context, tenan
 	}
 }
 
-func (ns *notificationsService) SendNotification(ctx context.Context, notification *webhookclient.NotificationRequest) (*webhookdir.Response, error) {
-	if notification == nil {
+func (ns *notificationsService) GenerateFormationNotifications(ctx context.Context, formationTemplateWebhooks []*model.Webhook, tenantID string, formation *model.Formation, formationTemplateID string, formationOperation model.FormationOperation) ([]*webhookclient.FormationNotificationRequest, error) {
+	if len(formationTemplateWebhooks) == 0 {
+		log.C(ctx).Infof("Formation template with ID: %q does not have any webhooks", formationTemplateID)
 		return nil, nil
 	}
-	resp, err := ns.webhookClient.Do(ctx, notification)
+
+	log.C(ctx).Infof("There are %d formation template(s) listening for formation lifecycle notifications", len(formationTemplateWebhooks))
+
+	customerTenantContext, err := ns.extractCustomerTenantContext(ctx, tenantID)
+	if err != nil {
+		return nil, errors.Wrapf(err, "while extracting customer tenant context for tenant with internal ID %s", tenantID)
+	}
+
+	formationTemplateInput := buildFormationLifecycleInput(formationOperation, formation, customerTenantContext)
+
+	requests := make([]*webhookclient.FormationNotificationRequest, 0, len(formationTemplateWebhooks))
+	for _, webhook := range formationTemplateWebhooks {
+		gqlWebhook, err := ns.webhookConverter.ToGraphQL(webhook)
+		if err != nil {
+			return nil, errors.Wrapf(err, "while converting formation template webhook with ID: %s to graphql one", webhook.ID)
+		}
+
+		req := &webhookclient.FormationNotificationRequest{
+			Request: webhookclient.NewRequest(
+				*gqlWebhook,
+				formationTemplateInput,
+				correlation.CorrelationIDFromContext(ctx),
+			),
+		}
+
+		requests = append(requests, req)
+	}
```

Comment:
This needs to be extracted in the notification builder and have the respective EnforceConstraints plugged in it.

Reply:
Addressed with https://github.com/kyma-incubator/compass/pull/2916